src/sql/base/parts/tree/browser/treeModel.ts

```diff
@@ -525,12 +533,34 @@ export class Item {
 
 			return result
 				.then(undefined, onUnexpectedError)
-				.then(() => this._onDidRefreshChildren.fire(eventData));
+				.then(() => this._onDidRefreshChildren.fire(eventData))
+				.then(() => {
+					// All children have been refreshed, if object is original item to refresh, we can complete refresh.
+					if (this === Item.originalItem) {
+						this.loadCompleteSpinnerWheel();
+					}
+				});
 		};
 
 		return safe ? doRefresh() : this.lock.run(this, doRefresh);
 	}
 
+	private loadCompleteSpinnerWheel(): void {
+		/**
+		 * Refresh loading status for item will be instantly lost upon refresh of item self
+		 * (usually before the spinner shows). we need to temporarily bring it back upon
+		 * actual refresh completion to let the user know the refresh happened with a short
+		 * appearance of the loading spinner.
+		 */
+		Item.originalItem?.addTrait('loading');
+		this.refreshLoadTimeout = setTimeout(() => {
+			Item.originalItem?.removeTrait('loading');
+			// Reset original item for next refresh.
+			Item.originalItem = undefined;
+			this.refreshLoadTimeout = undefined;
+		}, 1000);
```

Comment:
So you're still doing that "show it for 1sec regardless of how long it actually takes to fully refresh"? That still doesn't seem like the solution we want here - if we're going to fix this it seems like something we should actually fix completely so it shows the loading spinner while it's refreshing and not any more/less. 

(We don't want to be in a situation where the refresh takes longer and the spinner disappears but then the refresh isn't quite finished. Same but to a slightly lesser extent for refresh being quicker than the timeout)

Reply:
I'm just replicating the behavior I found on the extensions, which also use a timeout to display the loading bar even after they finish loading. Also, I found out one of the listeners for one of the events here already removes the loading spinner upon completion of the refresh. If users won't see the refresh icon at all, would that be fine for them?

Reply:
I can remove the timeout part of the function, but users won't likely see anything if this is the case, for most ordinary things,